csvGenerator.ipynb

```diff
@@ -0,0 +1,152 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "6f275807",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import csv\n",
+    "import math\n",
+    "import random\n",
+    "import requests\n",
+    "from CitesphereConnectorTest import *\n",
+    "from CitesphereConnector import CitesphereConnector"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "2c0597b1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "auth_object = authObject()\n",
+    "auth_object.authType  = 'oauth'\n",
+    "auth_object.access_token=\"b7f4880b-7e33-4eca-b5a5-29c73adce87a\"\n",
+    "connector = CitesphereConnector(\"https://diging-dev.asu.edu/citesphere-review/api\", auth_object)\n",
+    "groups=connector.get_groups()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "37167759",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def downloadFile(file_id, citesphere_token):\n",
+    "    giles_url = f\"https://diging.asu.edu/geco-giles-staging/api/v2/resources/files/{file_id}/content\"\n",
+    "    folder_path = \"citesphere-connector\"\n",
+    "    os.makedirs(folder_path, exist_ok=True)\n",
+    "    filename = os.path.join(folder_path, f\"{file_id}.pdf\")\n",
+    "    headers = {\n",
+    "        \"Authorization\": f\"Bearer {citesphere_token}\",\n",
+    "        \"Content-Type\": \"application/pdf;charset=UTF-8\"\n",
+    "    }\n",
+    "    response = requests.get(giles_url, headers=headers)\n",
+    "    if response.status_code == 200:\n",
+    "        with open(filename, \"wb\") as file:\n",
+    "            file.write(response.content)\n",
+    "        return filename  \n",
+    "    else:\n",
+    "        return None  \n",
+    "\n",
+    "def addToCSV(itm:list,csv_dict:dict):\n",
+    "    for value in itm[\"items\"]:\n",
```

Comment:
if `itm` is a list, the name should indicate that. also it's misspelled (should be "item" not "itm")

Reply:
added hints to all functions